lib/jellyfish/config_reader.ex

```diff
@@ -128,12 +103,121 @@ defmodule Jellyfish.ConfigReader do
+        cookie = parse_cookie(cookie_value)
+        nodes = parse_nodes(nodes_value)
+
+        if nodes == [] do
+          Logger.warning("""
+          JF_DIST_ENABLED has been set but JF_DIST_NODES remains unset.
+          Make sure that at least one of your Jellyfish instances
+          has JF_DIST_NODES set.
+          """)
+        end
+
+        [
+          enabled: true,
+          strategy: Cluster.Strategy.Epmd,
+          node_name: node_name,
+          cookie: cookie,
+          strategy_config: [hosts: nodes]
+        ]
+
+      dist_strategy == "DNS" ->
+        do_read_dns_config(dist_strategy, node_name_value, cookie_value)
+
+      true ->
+        raise """
+        JF_DIST_ENABLED has been set but unknown JF_DIST_STRATEGY was provided.
+        Availabile strategies are EPMD or DNS, provided strategy name was: "#{dist_strategy}"
+        """
+    end
+  end
+
+  defp do_read_dns_config(dist_strategy, node_name_value, cookie_value) do
+    unless node_name_value do
+      raise "JF_DIST_ENABLED has been set but JF_DIST_NODE_NAME remains unset."
+    end
+
+    node_name = parse_node_name(node_name_value)
+    cookie = parse_cookie(cookie_value)
+
+    parse_dns_string = fn env_name ->
+      env = System.get_env(env_name)
+
+      if !env and dist_strategy == "DNS" do
+        raise "DNS strategy has been set but #{env_name} remains unset."
+      end
+
+      env
+    end
+
+    query = parse_dns_string.("JF_DIST_QUERY")
+
+    [node_basename, _ip_addres_or_fqdn | []] = String.split(node_name_value, "@")
+
+    polling_interval = parse_polling_interval()
+
+    [
+      enabled: true,
+      strategy: Cluster.Strategy.DNSPoll,
+      node_name: node_name,
+      cookie: cookie,
+      strategy_config: [
+        polling_interval: polling_interval,
+        query: query,
+        node_basename: node_basename
+      ]
+    ]
+  end
+
+  defp parse_node_name(node_name) do
+    case String.split(node_name, "@") do
+      [_node_basename, _ip_addres_or_fqdn | []] ->
+        String.to_atom(node_name)
+
+      _other ->
+        raise "JF_DIST_NODE_NAME has improper format should have <basename>@<ip_address> or <basename>@<fqdn> and has got: #{node_name}"
+    end
+  end
+
   defp parse_cookie(cookie_value), do: String.to_atom(cookie_value)
 
   defp parse_nodes(nodes_value) do
     nodes_value
     |> String.split(" ", trim: true)
     |> Enum.map(&String.to_atom(&1))
   end
+
+  defp parse_polling_interval() do
+    env_value = System.get_env("JF_DIST_POLLING_INTERVAL", "5000")
+
+    case Integer.parse(env_value) do
+      {polling_interval, ""} when polling_interval > 0 ->
+        polling_interval
+
+      {polling_interval, ""} ->
+        raise "`JF_DIST_POLLING_INTERVAL` must be positivie integer. Provided value was: #{polling_interval}"
+
+      _other ->
+        raise "Error during parsing `JF_DIST_POLLING_INTERVAL`. Provided value should be integer and was: #{env_value}"
```

Comment:
This should be enough:

```suggestion
      _other ->
        raise "`JF_DIST_POLLING_INTERVAL` must be a positivie integer. Got: #{env_value}"
```